code/game/machinery/kitchen/smartfridge.dm

```diff
@@ -156,17 +156,12 @@
 				var/obj/item/chems/food/snacks/S = thing
 				if(S.dry || !I.get_specific_product(get_turf(src), S))
 					continue
-				if(S.dried_type == S.type)
-					S.dry = 1
-					S.SetName("dried [S.name]")
-					S.color = "#a38463"
-					stock_item(S)
+				var/result = S.on_dry(get_turf(src))
```

Comment:
The drying refactor stuff could also be split into a separate PR, anything that improves the mess of a system we have currently is good.